lib/_tls_wrap.js

```diff
@@ -312,8 +324,12 @@ function TLSSocket(socket, opts) {
 
   var wrap;
   if ((socket instanceof net.Socket && socket._handle) || !socket) {
+    // 1. connected socket
+    // 2. no socket, one will be created with net.Socket().connect
     wrap = socket;
   } else {
+    // 3. socket has no handle so is js not c++
```

Comment:
`so it is`?

Reply:
made this change as it landed, thanks @indutny